packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -363,6 +363,11 @@ def add_block(self, block: Block) -> None:
         """Add a block to the list."""
         expected_height = self.height + 1
         actual_height = block.header.height
+        if actual_height < self._height_offset:
+            # if the current block has a lower height than the
+            # initial height, ignore it
+            return
```

Comment:
Should we only ignore these blocks if we are `x` secs after a hard reset?

Reply:
This solution doesnt take into consideration that.